internal/server/proto/server.proto

```diff
@@ -2144,16 +2146,33 @@ message Runner {
   // jobs from being assigned to them.
   bool by_id_only = 2;
 
-  // odr is true if this runner is an on-demand runner. For ODRs, we expect
-  // they will accept exactly one job and then exit. This is used by the
-  // server to change some other behavior:
-  //
-  //   * The server will give ODRs project/app-scoped config if it exists.
-  //   * The server will never assign more than one job to this runner.
-  //     This is also enforced in the runner client-side but the server also
-  //     does this out of caution.
-  //
-  bool odr = 4;
+  // Location indicates what kind of runner this is
+  oneof type {
```

Comment:
Happy to bikeshed on names here. I don't love `type`.

Reply:
We use `kind` for token types, maybe we could use that here too?

Reply:
Oh good idea